src/lexer.rs

```diff
@@ -12,27 +12,19 @@ use crate::FunctionIdentifier::{Cbrt, Ceil, Cos, Exp, Abs, Floor, Ln, Log, Round
 use crate::units::Unit;
 use crate::units::Unit::*;
 use unicode_segmentation::{Graphemes, UnicodeSegmentation};
+use std::cmp::Ordering;
 
 fn is_word_char_str(input: &str) -> bool {
-  let x = match input {
-    "A" | "B" | "C" | "D" | "E" | "F" | "G" | "H" | "I" | "J" | "K" | "L"
+  matches!(input, "A" | "B" | "C" | "D" | "E" | "F" | "G" | "H" | "I" | "J" | "K" | "L"
     | "M" | "N" | "O" | "P" | "Q" | "R" | "S" | "T" | "U" | "V" | "W" | "X"
-    | "Y" | "Z" => true,
-    "a" | "b" | "c" | "d" | "e" | "f" | "g" | "h" | "i" | "j" | "k" | "l"
+    | "Y" | "Z" | "a" | "b" | "c" | "d" | "e" | "f" | "g" | "h" | "i" | "j" | "k" | "l"
     | "m" | "n" | "o" | "p" | "q" | "r" | "s" | "t" | "u" | "v" | "w" | "x"
-    | "y" | "z" => true,
-    "Ω" | "Ω" | "µ" | "μ" => true,
-    _ => false,
-  };
-  return x;
+    | "y" | "z" | "Ω" | "Ω" | "µ" | "μ")
+
 }
 
 fn is_numeric_str(input: &str) -> bool {
-  match input {
-    "." => true,
-    "0" | "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9" => true,
-    _ => false,
-  }
+  matches!(input, "." | "0" | "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9")
```

Comment:
Not really liking the use of `matches!`, it hurts readability too much in my opinion

Reply:
Isn't matches! much more readable? It removes most of the match syntax (like => and the explicit true and false).

Reply:
It removes the grouping and is less explicit, so I would prefer to stick with the match statement